analysers/analyser_merge_milestone_FR_metropole.py

```diff
@@ -0,0 +1,98 @@
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         ##
+## GNU General Public License for more details.                          ##
+##                                                                       ##
+## You should have received a copy of the GNU General Public License     ##
+## along with this program.  If not, see <http://www.gnu.org/licenses/>. ##
+##                                                                       ##
+###########################################################################
+
+from .Analyser_Merge import Analyser_Merge, Source, CSV, Load, Mapping, Select, Generate
+
+class Analyser_Merge_Milestone_FR_metropole(Analyser_Merge):
+    def __init__(self, config, logger = None):
+        Analyser_Merge.__init__(self, config, logger)
+
+        doc = dict(
+            detail = T_(
+'''The list of milestone comes from the CEREMA's database "RIU" in France.'''),
+            trap = T_(
+'''The position of the marker may be a little different than that visible on the road. Sometimes, a small white line perpendicular to the road on the emergency stop strip or the left flared strip can be seen on satellite images or Mapillary's photo.'''))
+
+        self.def_class_missing_official(item = 8430, id = 41, level = 3, tags = ['merge', 'highway'],
+            title = T_('Milestone not integrated'), **doc)
+        self.def_class_possible_merge(item = 8431, id = 43, level = 3, tags = ['merge', 'highway'],
+            title = T_('Milestone integration suggestion'), **doc)
+        self.def_class_update_official(item = 8432, id = 44, level = 3, tags = ['merge', 'highway'],
+            title = T_('Milestone update'), **doc)
+
+        self.init(
+            "https://www.data.gouv.fr/fr/datasets/bornage-du-reseau-routier-national/",
+            "Bornage du réseau routier national",
+            CSV(Source(attribution = "data.gouv.fr:Ministère de la Transition écologique et solidaire", millesime = "01/2019",
+                    fileUrl = "https://www.data.gouv.fr/fr/datasets/r/fbc8b73b-a65c-486b-a710-ed22b9e4070c"),
+                    separator = "\t"),
+            Load("x", "y", srid = 2154,
+                xFunction = self.float_comma,
+                yFunction = self.float_comma,
+                where = lambda row: self.is_natref(row)),
+            Mapping(
+                select = Select(
+                    types = ["nodes"],
+                    tags = [{"highway": "milestone"}]),
+                osmRef = "nat_ref",
+                extraJoin = "ref",
+                conflationDistance = 150,
+                generate = Generate(
+                    static1 = {"highway": "milestone"},
+                    static2 = {"source": self.source},
+                    mapping1 = {
+                        "distance": 'pr',
+                        "nat_ref": lambda row: self.transform_to_natref(row),
+                        "ref": lambda row: self.transform_route(row['route']) }
```

Comment:
According the wiki the `ref` should be the ref of the milestone not the road, shouldn't?

Reply:
there is 2 information visible on a milestone, ex :
N 999 written in white on a red background and below 33 written in black on a white background. => distance = 33 and ref = N 999.
(nat_ref is not visible)

Reply:
But, N 999 is the ref of the road, not the one of the milestone. Maybe we have no ref visible in France?

Reply:
for me there is no ambiguity, it may be because of the way the national road network is described

Reply:
I look around in some country and I found sometime the ref is the distance, sometime the highway ref, or none.
It is not so obvious to me, as a ref should be unique under a context, else it is not a ref.

Reply:
I don't know what to answer or do ...

Reply:
Try to report this question to talk-fr

Reply:
thanks, it's done

Reply:
The nat_ref also is not uniq?

```
      4 <tag action="create" k="nat_ref" v="33PR18D" />
      4 <tag action="create" k="nat_ref" v="33PR18G" />
      4 <tag action="create" k="nat_ref" v="33PR19D" />
      4 <tag action="create" k="nat_ref" v="33PR19G" />
      4 <tag action="create" k="nat_ref" v="33PR35DC" />
      4 <tag action="create" k="nat_ref" v="33PR35GC" />
      4 <tag action="create" k="nat_ref" v="33PR36DC" />
      4 <tag action="create" k="nat_ref" v="33PR36GC" />
      4 <tag action="create" k="nat_ref" v="33PR37DC" />
      4 <tag action="create" k="nat_ref" v="33PR37GC" />
      4 <tag action="create" k="nat_ref" v="33PR38DC" />
      4 <tag action="create" k="nat_ref" v="33PR38GC" />
      4 <tag action="create" k="nat_ref" v="33PR39DC" />
      4 <tag action="create" k="nat_ref" v="33PR39GC" />
      4 <tag action="create" k="nat_ref" v="33PR40DC" />
      4 <tag action="create" k="nat_ref" v="33PR40GC" />
      5 <tag action="create" k="nat_ref" v="33PR1D" />
      5 <tag action="create" k="nat_ref" v="33PR1G" />
      5 <tag action="create" k="nat_ref" v="33PR2D" />
      5 <tag action="create" k="nat_ref" v="33PR2G" />
      5 <tag action="create" k="nat_ref" v="33PR3D" />
      5 <tag action="create" k="nat_ref" v="33PR3G" />
      5 <tag action="create" k="nat_ref" v="33PR4D" />
      5 <tag action="create" k="nat_ref" v="33PR4G" />
      5 <tag action="create" k="nat_ref" v="33PR7D" />
      5 <tag action="create" k="nat_ref" v="33PR7G" />
      5 <tag action="create" k="nat_ref" v="33PR8D" />
      5 <tag action="create" k="nat_ref" v="33PR8G" />
      5 <tag action="create" k="nat_ref" v="33PR9D" />
      5 <tag action="create" k="nat_ref" v="33PR9G" />
      6 <tag action="create" k="nat_ref" v="33PR0D" />
      6 <tag action="create" k="nat_ref" v="33PR0G" />
      6 <tag action="create" k="nat_ref" v="33PR5D" />
      6 <tag action="create" k="nat_ref" v="33PR5G" />
      6 <tag action="create" k="nat_ref" v="33PR6D" />
      6 <tag action="create" k="nat_ref" v="33PR6G" />
```

Reply:
no, uniqueness is only for a route
(non, l'unicité est uniquement pour une route, cf page 13 DT5792.pdf)

Reply:
as there is not issue in talk-fr :
- do the job with only nat_ref ?
- do nothing because I don't see any interest in data integration without both rrn "route" and "plo" data.

Reply:
I think actual nat_ref should go to ref (or unsigned_ref)
It would nice to have the road ref in some tag.

Reply:
the different comments on the list make me think that there is no longer any reason to use ref, nat_ref or other osm tags, but rather specific tags which are not ambiguous like :
ref:FR:rrn:route=row["route"]
ref:FR:rrn:plo=transform_to_natref
(and modify the name of this function to transform_to_plo)

Reply:
Yes, better.

Reply:
Can you make this change (or an other) so we can merge it?